src/dist/app_daemon/daemon.server.cpp

```diff
@@ -111,39 +111,12 @@ namespace dsn
             }
         }
 
-        static std::vector<rpc_address> read_meta_servers()
-        {
-            std::vector<rpc_address> meta_servers;
-
-            const char* server_ss[10];
-            int capacity = 10, need_count;
-            need_count = dsn_config_get_all_keys("meta_servers", server_ss, &capacity);
-            dassert(need_count <= capacity, "too many meta servers specified");
-
-            std::ostringstream oss;
-            for (int i = 0; i < capacity; i++)
-            {
-                std::string s(server_ss[i]);
-                // name:port
-                auto pos1 = s.find_first_of(':');
-                if (pos1 != std::string::npos)
-                {
-                    ::dsn::rpc_address ep(s.substr(0, pos1).c_str(), atoi(s.substr(pos1 + 1).c_str()));
-                    meta_servers.push_back(ep);
-                    oss << "[" << ep.to_string() << "] ";
-                }
-            }
-            ddebug("read meta servers from config: %s", oss.str().c_str());
-
-            return std::move(meta_servers);
-        }
-
-
         DEFINE_TASK_CODE(LPC_DAEMON_APPS_CHECK_TIMER, TASK_PRIORITY_COMMON, THREAD_POOL_DEFAULT)
 
         void daemon_s_service::open_service()
         {
-            auto meta_servers = std::move(read_meta_servers());
+            std::vector<rpc_address> meta_servers;
+            dsn::replication::replica_helper::load_meta_servers(meta_servers);
```

Comment:
wrong dependency of daemon atop replica_helper... that's why we have redundancies.